tests/unit_tests/test_helm_chart.py

```diff
@@ -96,81 +96,18 @@ def test_helm_chart_creates_config_map(worker_config: ApplicationConfig):
     assert rendered_config == worker_config
 
 
-@pytest.mark.parametrize(
-    "values",
-    [
-        {
-            "initContainer": {
-                "enabled": True,
-                "scratch": {
-                    "repositories": [],
-                    "root": "/blueapi-plugins/scratch",
-                },
-            }
-        },
-        {
-            "initContainer": {
-                "enabled": True,
-                "scratch": {
-                    "root": "/dls_sw/i22/scratch",
-                    "required_gid": 12345,
-                    "repositories": [
-                        {
-                            "name": "foo",
-                            "remote_url": "https://example.git",
-                        },
-                        {
-                            "name": "bar",
-                            "remote_url": "https://example.git",
-                        },
-                    ],
-                },
-            }
-        },
-    ],
-)
-def test_helm_chart_creates_init_config_map(values: Values):
-    manifests = render_chart(values=values)
-    rendered_config = yaml.safe_load(
-        manifests["ConfigMap"]["blueapi-initconfig"]["data"]["initconfig.yaml"]
-    )
-    assert rendered_config == values["initContainer"]
```

Comment:
This test would have caught the `enabled` issue, so I'd like to see a version of it reinstatated that turns init-config.yaml into an ApplicationConfig to check that no extra fields creep in.

Reply:
See comment on `test_init_container_config_copied_from_worker_when_enabled`: I would remove that test and move the logic into this, just to remove the size of the diff